test/CMakeLists.txt

```diff
@@ -7,43 +7,62 @@ target_include_directories(${PROJECT_NAME}_TestPlugins1
   PUBLIC "../include" ${console_bridge_INCLUDE_DIRS} ${Poco_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME}_TestPlugins1 ${PROJECT_NAME})
 class_loader_hide_library_symbols(${PROJECT_NAME}_TestPlugins1)
-
+if(WIN32)
+target_compile_definitions(${PROJECT_NAME}_TestPlugins1 PRIVATE "CLASS_LOADER_BUILDING_DLL")
```

Comment:
My expectation would be that this should only be defined when `class_loader` is being built. Why is this necessary here? And if it is I would suggest to use a separate name to indicate that the "using" part of class loader is being built here.

Reply:
Without this, the tests linking against these testing libraries fail to link.

>  And if it is I would suggest to use a separate name to indicate that the "using" part of class loader is being built here.

I'm not sure I get what you mean here, do you suggest creating another set of visibility macros for the libraries built and used for testing?